### PR TITLE
Feature/automation timer trigger 2

### PR DESCRIPTION
**Pull request by chriskn**

### CODE REVIEW

File: bundles/automation/org.eclipse.smarthome.automation.module.core.test/pom.xml [hunk 1]
```diff
@@ -27,6 +27,11 @@
 					<dependencies>
 						<dependency>
 							<type>eclipse-plugin</type>
+							<artifactId>org.eclipse.smarthome.core.scheduler</artifactId>
```
Comment (BenediktNiehues) [inline bundles/automation/org.eclipse.smarthome.automation.module.core.test/pom.xml]: this should be removed again as the timer module is not part of module.core anymore

File: bundles/automation/org.eclipse.smarthome.automation.module.core.test/pom.xml [hunk 2]
```diff
@@ -94,6 +99,11 @@
 					</dependencies>
 					<bundleStartLevel>
 						<bundle>
+							<id>org.eclipse.smarthome.core.scheduler</id>
```
Comment (BenediktNiehues) [inline bundles/automation/org.eclipse.smarthome.automation.module.core.test/pom.xml]: same here

File: bundles/automation/org.eclipse.smarthome.automation.module.core.test/src/test/groovy/org/eclipse/smarthome/automation/module/core/internal/RuntimeRuleTest.groovy [hunk 3]
```diff
@@ -201,50 +204,96 @@ class RuntimeRuleTest extends OSGiTest{
             assertThat mtr.get(CompareConditionHandler.MODULE_TYPE), is(notNullValue())
         },3000,100)
     }
-    
+
+    @Test
```
Comment (BenediktNiehues) [inline bundles/automation/org.eclipse.smarthome.automation.module.core.test/src/test/groovy/org/eclipse/smarthome/automation/module/core/internal/RuntimeRuleTest.groovy:205]: tests moved to the timer.test bundle, right?

File: bundles/automation/org.eclipse.smarthome.automation.module.core.test/src/test/groovy/org/eclipse/smarthome/automation/module/core/internal/RuntimeRuleTest.groovy [hunk 4]
```diff
@@ -75,7 +76,8 @@ class RuntimeRuleTest extends OSGiTest{
                     new SwitchItem("myLampItem3"),
                     new SwitchItem("myMotionItem4"),
                     new SwitchItem("myPresenceItem4"),
-                    new SwitchItem("myLampItem4")
+                    new SwitchItem("myLampItem4"),
+                    new SwitchItem("myLampItem5")
```
Comment (BenediktNiehues) [inline bundles/automation/org.eclipse.smarthome.automation.module.core.test/src/test/groovy/org/eclipse/smarthome/automation/module/core/internal/RuntimeRuleTest.groovy]: if it is just used for timer test, this has to be removed as well

File: bundles/automation/org.eclipse.smarthome.automation.module.core/ESH-INF/automation/moduletypes/EventTriggersTypeDefinition.json [hunk 5]
```diff
@@ -74,6 +74,20 @@
                "reference":"itemStateChangeTriggerID.event"
             }
          ]
+      },
```
Comment (BenediktNiehues) [inline bundles/automation/org.eclipse.smarthome.automation.module.core/ESH-INF/automation/moduletypes/EventTriggersTypeDefinition.json]: moved to module.timer....

File: bundles/automation/org.eclipse.smarthome.automation.module.core/META-INF/MANIFEST.MF [hunk 6]
```diff
@@ -18,8 +18,12 @@ Import-Package: com.google.common.collect,
  org.osgi.framework,
  org.osgi.service.event,
  org.osgi.util.tracker,
+ org.quartz;resolution:=optional,
```
Comment (BenediktNiehues) [inline bundles/automation/org.eclipse.smarthome.automation.module.core/META-INF/MANIFEST.MF]: not needed anymore

File: bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/factory/BasicModuleHandlerFactory.java [hunk 7]
```diff
@@ -62,7 +63,7 @@ public BasicModuleHandlerFactory(BundleContext bundleContext) {
 
     @SuppressWarnings({ "unchecked", "rawtypes" })
     private void initializeServiceTrackers() {
-        this.itemRegistryTracker = new ServiceTracker(this.bundleContext, ItemRegistry.class.getName(),
+        this.itemRegistryTracker = new ServiceTracker(this.bundleContext, ItemRegistry.class,
```
Comment (BenediktNiehues) [inline bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/factory/BasicModuleHandlerFactory.java]: ItemRegistry.class.getName() !!

File: bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/factory/BasicModuleHandlerFactory.java [hunk 8]
```diff
@@ -83,7 +84,7 @@ public void removedService(ServiceReference reference, Object service) {
                 });
         this.itemRegistryTracker.open();
 
-        this.eventPublisherTracker = new ServiceTracker(this.bundleContext, EventPublisher.class.getName(),
+        this.eventPublisherTracker = new ServiceTracker(this.bundleContext, EventPublisher.class,
```
Comment (BenediktNiehues) [inline bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/factory/BasicModuleHandlerFactory.java]: EventPublisher.class.getName()!!

File: bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/factory/BasicModuleHandlerFactory.java [hunk 9]
```diff
@@ -108,6 +109,13 @@ public void removedService(ServiceReference reference, Object service) {
 
     @Override
     public Collection<String> getTypes() {
+        try {
```
Comment (BenediktNiehues) [inline bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/factory/BasicModuleHandlerFactory.java]: can be removed

File: bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/factory/BasicModuleHandlerFactory.java [hunk 10]
```diff
@@ -231,14 +238,26 @@ protected ModuleHandler internalCreate(Module module, String ruleUID) {
             CompareConditionHandler compareConditionHandler = handler != null
                     && handler instanceof CompareConditionHandler ? (CompareConditionHandler) handler : null;
             if (compareConditionHandler == null) {
-                compareConditionHandler = new CompareConditionHandler((Condition)module);
-                handlers.put(ruleUID+module.getId(), compareConditionHandler);
+                compareConditionHandler = new CompareConditionHandler((Condition) module);
+                handlers.put(ruleUID + module.getId(), compareConditionHandler);
             }
             return compareConditionHandler;
+        } else if (TimerTriggerHandler.MODULE_TYPE_ID.equals(moduleTypeUID) && module instanceof Trigger) {
```
Comment (BenediktNiehues) [inline bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/factory/BasicModuleHandlerFactory.java]: remove

File: bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/handler/CallbackJob.java [hunk 11]
```diff
@@ -0,0 +1,58 @@
+/**
```
Comment (BenediktNiehues) [inline bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/handler/CallbackJob.java]: delete

File: bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/handler/TimerTriggerHandler.java [hunk 12]
```diff
@@ -0,0 +1,87 @@
+/**
```
Comment (BenediktNiehues) [inline bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/handler/TimerTriggerHandler.java]: delete if moved

File: bundles/automation/org.eclipse.smarthome.automation.module.timer.test/ESH-INF/automation/Xrules/ItemRuleDefinition.json [hunk 13]
```diff
@@ -0,0 +1,83 @@
+[
```
Comment (BenediktNiehues) [inline bundles/automation/org.eclipse.smarthome.automation.module.timer.test/ESH-INF/automation/Xrules/ItemRuleDefinition.json]: can be deleted

File: bundles/automation/org.eclipse.smarthome.automation.module.timer.test/META-INF/MANIFEST.MF [hunk 14]
```diff
@@ -0,0 +1,44 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: Eclipse SmartHome Automation Timer Test
+Bundle-SymbolicName: org.eclipse.smarthome.automation.module.timer.test;singleton:=true
+Bundle-Version: 0.8.0.qualifier
+Bundle-Vendor: Eclipse/SmartHome
+Bundle-RequiredExecutionEnvironment: JavaSE-1.7
+Import-Package: com.google.common.collect,
+ groovy.lang,
+ org.codehaus.groovy.reflection,
```
Comment (BenediktNiehues) [inline bundles/automation/org.eclipse.smarthome.automation.module.timer.test/META-INF/MANIFEST.MF:10]: check if all imports are really needed here!

**Comment by BenediktNiehues** [comment]
Don't forget to add the new bundles as module to parent pom!

**Comment by chriskn** [comment]
Updates related to comments

**Pull request merged**